OpenRA.Mods.Common/Widgets/Logic/Editor/SaveMapLogic.cs

```diff
@@ -93,10 +93,16 @@ public SaveMapLogic(Widget widget, Action<string> onSave, Action onExit, Map map
 					directoryDropdown.ShowDropDown("LABEL_DROPDOWN_TEMPLATE", 210, mapDirectories.Keys, setupItem);
 			}
 
-			var attr = File.GetAttributes(map.Path);
-			var mapIsUnpacked = attr.HasFlag(FileAttributes.Directory);
+			var mapIsUnpacked = false;
+
+			if (map.Path != null)
+			{
+				var attr = File.GetAttributes(map.Path);
+				mapIsUnpacked = attr.HasFlag(FileAttributes.Directory);
+			}
+
 			var filename = widget.Get<TextFieldWidget>("FILENAME");
-			filename.Text = mapIsUnpacked ? Path.GetFileName(map.Path) : Path.GetFileNameWithoutExtension(map.Path);
+			filename.Text = mapIsUnpacked && map.Path != null ? Path.GetFileName(map.Path) : Path.GetFileNameWithoutExtension(map.Path);
```

Comment:
It won't safeguard anything, `map.Path` is used on both sides of the `:`. Also both `Path.GetFilename` and `Path.GetFileNameWithoutExtension` work fine when you pass them a `null`. Please remove it.